Drivers/Src/CANbus.c

```diff
@@ -39,18 +39,29 @@ static void CANbus_CountIncoming(void) {
 /**
  * @brief   Initializes the CAN system
  * @param   loopback	: if we should use loopback mode (for testing)	
+ * @param 	faultState  : determines whether to implement Rx and Tx interrupts
  * @return  None
  */
-void CANbus_Init(bool loopback) {
+void CANbus_Init(bool loopback, bool faultState) {
+	if(!faultState){
+		RTOS_BPS_MutexCreate(&CANbus_TxMutex, "CAN TX Lock");
+		RTOS_BPS_MutexCreate(&CANbus_RxMutex, "CAN RX Lock");
+		RTOS_BPS_SemCreate(&CANbus_MailSem4, "CAN Mailbox Semaphore", 3); // # of mailboxes
+		RTOS_BPS_SemCreate(&CANbus_ReceiveSem4, "CAN Queue Counter Semaphore", 0);
+	}
 
-	RTOS_BPS_MutexCreate(&CANbus_TxMutex, "CAN TX Lock");
-	RTOS_BPS_MutexCreate(&CANbus_RxMutex, "CAN RX Lock");
-	RTOS_BPS_SemCreate(&CANbus_MailSem4, "CAN Mailbox Semaphore", 3); // # of mailboxes
-	RTOS_BPS_SemCreate(&CANbus_ReceiveSem4, "CAN Queue Counter Semaphore", 0);
 	// Initialize and pass interrupt hooks
     BSP_CAN_Init(CANbus_CountIncoming, CANbus_Release, loopback);
```

Comment:
I guess it's ok that ```CANbus_CountIncoming()``` and ```CANbus_Release()``` are RTOS-based functions for fault state since interrupts are disabled so this doesn't get called, but it would be nice to add a comment for clarification since it looks like it could be RTOS functions called during fault state

Alternatively pass in NULL if fault state to make things more clear

Reply:
fixed